src/test/java/test/reflect/ReflectionRecipesTest.java

```diff
@@ -0,0 +1,357 @@
+      Assert.assertNotNull(methodParameters);
+      final Parameter[] filteredParameters =
+        ReflectionRecipes.filter(methodParameters, InjectableParameter.Assistant.ALL_INJECTS);
+      Assert.assertNotNull(filteredParameters);
+
+      for (final Object[] expect : expected) {
+        Assert.assertNotNull(expect);
+        Assert.assertEquals(expect.length, 2);
+        Assert.assertNotNull(expect[0]);
+        Assert.assertTrue(expect[0] instanceof Object[]);
+        Assert.assertNotNull(expect[1]);
+        Assert.assertTrue(expect[1] instanceof Boolean);
+        Assert.assertEquals(
+          ReflectionRecipes.matchArrayEnding(filteredParameters, (Object[]) expect[0]),
+          expect[1]
+        );
+      }
+    }
+  }
+
+  @Test(dataProvider = "exactMatchDP")
+  public void exactMatchTest(final Object[][] expected, @NoInjection final Method method) {
+    if (expected != null) {
+      final Parameter[] methodParameters = ReflectionRecipes.getMethodParameters(method);
+      Assert.assertNotNull(methodParameters);
+      final Parameter[] filteredParameters =
+        ReflectionRecipes.filter(methodParameters, InjectableParameter.Assistant.ALL_INJECTS);
+      Assert.assertNotNull(filteredParameters);
+
+      for (final Object[] expect : expected) {
+        Assert.assertNotNull(expect);
+        Assert.assertEquals(expect.length, 2);
+        Assert.assertNotNull(expect[0]);
+        Assert.assertTrue(expect[0] instanceof Object[]);
+        Assert.assertNotNull(expect[1]);
+        Assert.assertTrue(expect[1] instanceof Boolean);
+        Assert.assertEquals(
+          ReflectionRecipes.exactMatch(filteredParameters, (Object[]) expect[0]),
+          expect[1]
+        );
+      }
+    }
+  }
+
+  @Test(dataProvider = "methodInputParameters")
+  public void testFilters(final Parameter[] parameters) {
+    System.out.println("In: " + Arrays.asList(parameters));
+    final Parameter[] parameters1 = ReflectionRecipes.filter(parameters, InjectableParameter.Assistant.ALL_INJECTS);
+    System.out.println("Out: " + Arrays.asList(parameters1));
+    Assert.assertEquals(parameters1.length, 2);
+    Assert.assertEquals(parameters1[0].getType(), int.class);
+    Assert.assertEquals(parameters1[1].getType(), Boolean.class);
+  }
+
+  @Test(dataProvider = "methodInputParamArgsPair")
+  public void testInject(final Parameter[] parameters, final Object[] args) {
+    System.out.println("In: " + Arrays.asList(parameters));
+    System.out.println("args: " + Arrays.asList(args));
+    try {
+      final Object[] injectedArgs = ReflectionRecipes.inject(
+        parameters, InjectableParameter.Assistant.ALL_INJECTS, args,
+        null, null, null
+      );
+      System.out.println("injectedArgs: " + Arrays.asList(injectedArgs));
+      Assert.assertEquals(injectedArgs.length, parameters.length);
+    }
+    catch (final Throwable throwable) {
+      throwable.printStackTrace();
```

Comment:
I think `Assert.fail()` is missing here. Otherwise, the test will never fail in case of exception.